### ISSUE TITLE
Handling correlated datasets

### ISSUE THREAD

**Issue by nemtiax** [google/automl#728]
I'm working with a dataset that consists of successive frames from a video feed.  Unlike COCO or VOC, adjacent images in the dataset are highly correlated and in many cases almost identical.  Further, the dataset contains contiguous stretches of time where no objects are present, and then contiguous stretches of time where one or more objects are present.  

It seems that the current code base only does a small amount of shuffling - it looks like dataloader shuffles local blocks of 64 items when building batches:

https://github.com/google/automl/blob/d2aaf023c7fb8f2d3f3a35f04aa3fac9343152f2/efficientdet/dataloader.py#L409

This is sufficient for COCO and VOC, since each COCO image has nothing to do with the one before it.  But for my data it obviously creates problems.  To resolve it locally, I added a call to random.shuffle() to create_coco_tfrecord here:

https://github.com/google/automl/blob/07f10f23c95753b23ecb482097aabdaffe8b6741/efficientdet/dataset/create_coco_tfrecord.py#L291

It seems like the sort of thing that other people might get bitten by, and would maybe be good to either add this shuffle automatically, or to add a flag to the tfrecord creation script(s) to let users request it.  I'm not sure if adding it automatically would maybe break some tests that expect deterministic behavior, or if there might be a better place to do this fix.  I assume that trying to shuffle after the tfrecords are created would be prohibitive as you'd have to read in a whole bunch of data before shuffling.

**Comment by kartik4949** [comment]
@nemtiax this shuffling is for shards ,so lets say you have one tfrecord with 32 shards ,so these 32 shards will be shuffled and for them 64 is enough and for shuffling tfrecord we have list_dir with shuffle true which does the job of shuffling the records

**Comment by nemtiax** [comment]
Will I still have a problem if my data is in sorted order within shards?  I certainly seemed to be getting very clear patterns in my training loss that corresponded to the ordering of my data, and adding the shuffle where I did seemed to remove those patterns entirely.

**Comment by kartik4949** [comment]
within shards , we have shuffling right ,so it will do the job
